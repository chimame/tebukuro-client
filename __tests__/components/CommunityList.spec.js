import React                  from 'react'
import { shallow }            from 'enzyme'
import toJson                 from 'enzyme-to-json'
import CommunityListComponent from '../../client/components/CommunityList'
import CommunityListModel     from '../../client/models/CommunityList'
import CommunityModel         from '../../client/models/Community'
import Params                 from '../factories/Community'

describe('CommunityList', () => {
  test('render list', () => {
    const communities = [new CommunityModel(Params.community1), new CommunityModel(Params.community2)]
    const list = new CommunityListModel({communities: communities})
    const component = shallow(
      <CommunityListComponent CommunityList={list} />
    )

    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
