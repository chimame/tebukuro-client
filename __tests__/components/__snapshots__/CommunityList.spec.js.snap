// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityList render list 1`] = `
<table>
  <thead>
    <tr>
      <th>
        name
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        name1
      </td>
    </tr>
    <tr>
      <td>
        name2
      </td>
    </tr>
  </tbody>
</table>
`;
